mautic/mautic
# Removing emoji transformer

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

The EmojiTransformer is causing the API Library tests to [fail](https://github.com/mautic/api-library/actions/runs/5254032028/jobs/9492132737) with issues like this:

```
Mautic\Tests\Api\DynamicContentsTest::testGetListOfSpecificIds
The response has unexpected status code (500).

Response: {"errors":[{"message":"Mautic\\CoreBundle\\Helper\\EmojiHelper::toShort(): Argument #1 ($text) must be of type string, null given, called in \/var\/www\/html\/app\/bundles\/CoreBundle\/Form\/DataTransformer\/EmojiToShortTransformer.php on line 47","code":500,"type":null}],"trace":[{"namespace":"","short_class":"","class":"","type":"","function":"","file":"\/var\/www\/html\/app\/bundles\/CoreBundle\/Helper\/EmojiHelper.php","line":31,"args":[]},{"namespace":"Mautic\\CoreBundle\\Helper","short_class":"EmojiHelper","class":"Mautic\\CoreBundle\\Helper\\EmojiHelper","type":"::","function":"toShort","file":"\/var\/www\/html\/app\/bundles\/CoreBundle\/Form\/DataTransformer\/EmojiToShortTransformer.php","line":47,"args":[]},{"namespace":"Mautic\\CoreBundle\\Form\\DataTransformer","short_class":"EmojiToShortTransformer","class":"Mautic\\CoreBundle\\Form\\DataTransformer\\EmojiToShortTransformer","type":"-\u003E","function":"reverseTransform","file":"\/var\/www\/html\/vendor\/symfony\/form\/Form...
Payload: {
    "name": "test",
    "content": "test"
}
Failed asserting that false is true.
```
We can fix the issue and leave the emoji transformer there, but I think it's time to remove the transformer completely. It was needed because older versions of MySql and encoding did not support all emojis. But since then we did changes to enable them:

- https://github.com/mautic/mautic/pull/10844
- https://github.com/mautic/mautic/pull/10354
- https://github.com/mautic/mautic/pull/9526
- https://github.com/mautic/mautic/pull/8349

So there is no need to transform emojis anymore. The emoji transformer also use hard-coded list of emojis that haven't been updated for years so all new emojis would fail anyway.

The only 2 fields where this transformer was used was in:
- Dynamic Web Content description field
- Email subject field

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a dynamic web content with no description
3. Create a dynamic web content with an emoji in the description
4. Create an email with an emoji in the subject

There should be no errors
